lib/instrumentation/modules/http2.js

```diff
@@ -178,6 +179,8 @@ module.exports = function (http2, agent, { enabled }) {
           url
         })
 
+        span.setDestinationContext(getHTTPDestination(url, span.type))
```

Comment:
🤔 I'm normally inclined to set stuff like this as early in the span lifecycle as possible. In this case just after the `url` variable as been created. That might help debugging later as the state of the span is as complete as possible as early as possible. On the other hand, this doesn't change the end result, so I'm 🤷‍♂️

Reply:
Dont have any strong preference either. Let me know what you think. I can update accordingly. Just followed the similarity with http context.